added simple root Event and fixed OGBillOfLading Generator

mkhraisha opened the pull request:
Adds a base "Event" Schema that captures very simple information. 

This schema should be extended by the following EPCIS event types:
* observation
* transformation
* aggregation
* disaggregation
* commission
* decommission

This PR is meant to spark conversation on what fields need to be in the base Event vs one of the Event Types

mprorock commented on packages/traceability-schemas/schemas/Event.json:
```diff
@@ -0,0 +1,67 @@
+            "description": "The organization performing the activity.",
+            "type": "array",
+            "items": {
+                "$ref": "https://w3id.org/traceability/schemas/Organization.json"
```
Do we want this to be an org? or is entity possibly better?  We at least also deal with individuals in the supply chain where Person sometimes is a better representation than Organization

mkhraisha replied:
I had misgivings on whether we should have two fields or just one for this, I wanted initially to have an `actor` and an `actingOrg` field. 

An entity might solve the problem. good shout

mprorock replied:
I would prefer use of [Entity](https://w3c-ccg.github.io/traceability-vocab/schemas/Entity.json)

mprorock commented on packages/traceability-schemas/schemas/Event.json:
```diff
@@ -0,0 +1,67 @@
+            "description": "Time when the event took place",
+            "type": "string"
+        },
+        "products": {
```
I am going to have to mull over use of Product here.  I think it is probably correct, but also thinking about raw materials becoming a product etc.

mprorock replied:
see note from @OR13 below - should this be GS1 or schema.org?  Other aspects of the vocab are utilizing schema.org

TallTed replied:
Consider "inputs" (raw materials, outputs from other processes/sources, etc.) and "outputs" (refined version of inputs becoming inputs to other processes/consumers, finished products, etc.).  These can then be easily turned into a complex *trace* from most-basic-origin (e.g., extraction of various ores from deep underground) to most-refined-consumable (e.g., a smartphone, tablet, or computer)...

mprorock commented on packages/traceability-schemas/schemas/Event.json:
```diff
@@ -0,0 +1,67 @@
+            "description": "Location where event took place",
+            "$ref": "https://w3id.org/traceability/schemas/Place.json"
+        },
+        "eventTime": {
```
we may want to actually record a few more things here (possibly optionally), e.g. scheduled or planned start time, planned duration, actual start time, actual end time, etc.  open on this.

mprorock replied:
I would prefer ability to record at a minimum

1. event start time
2. event end time

mkhraisha replied:
I think this makes sense to split up, probably another base object that encapsulates eventTime that would have the planned start/duration etc.

mprorock reviewed (changes requested):
Awesome start on this - I think the Event is going to be a a key interop point, and we will certianly need to issue VCs against certain events at least in our case on the ag side of things

OR13 commented:
@mprorock  can you clearly state what changes need to be made in order to accept this PR?

potentially we can open issues to track them and iterate?

OR13 commented on packages/traceability-schemas/schemas/Event.json:
```diff
@@ -0,0 +1,67 @@
+            }
+        },
+        "eventLocation": {
+            "$comment": "{\"term\": \"eventLocation\", \"@id\": \"https://w3id.org/traceability#eventLocation\"}",
```
I am not sure we want to be defining "eventLocation"... can we use an existing ontology for this field?

mprorock replied:
+1 I think this should be a [Place](https://w3c-ccg.github.io/traceability-vocab/#Place)

mkhraisha replied:
absolutely should be place

OR13 commented on packages/traceability-schemas/schemas/Event.json:
```diff
@@ -0,0 +1,67 @@
+            "type": "string"
+        },
+        "products": {
+            "$comment": "{\"term\": \"products\", \"@id\": \"https://www.gs1.org/voc/Product\"}",
```
noting the use of a GS1 vocabulary item here, which is close to but not equal to schema.org product.

mprorock replied:
> @mprorock can you clearly state what changes need to be made in order to accept this PR?
> 
> potentially we can open issues to track them and iterate?

noted above

OR13 commented on packages/traceability-schemas/src/generators/Event.js:
```diff
@@ -0,0 +1,27 @@
+const getEvent = () => {
+    const eventType = 'commission';
+    const eventId = '12345';
+    const actor = [getOrganization(), getOrganization()];
```
likely these helpers are injecting `@context`... please make sure we are not testing them in example data.

mkhraisha replied:
theres two ways to fix this in my mind either:
 * we need a parameter to these helpers that indicates whether to inject context or not (my preference)
 * we need to strip context after its generated in each of these helpers

OR13 reviewed (changes requested):
https://github.com/w3c-ccg/traceability-vocab/pull/164/files#r634604678

^ this is the only critical objection I have, we need to be cleaning up our examples and avoid context nesting.

mkhraisha commented:
Removed child context @mprorock  i deleted the generator and addressed most things in this conversation except for the event time stuff. would it be ok if i open an issue and assign it to you?

mprorock commented:
> event time stuff

let's not worry about the event time stuff as a blocker for now - we can add that as a later detail when required - probably as a subclass of this (e.g. event with duration or something like that)

OR13 commented:
On the call we discussed giving this one for day for reviews.

TallTed commented on packages/traceability-schemas/schemas/Event.json:
```diff
@@ -0,0 +1,67 @@
+    "$schema": "https://json-schema.org/draft-07/schema#",
+    "$comment": "{\"term\": \"Event\", \"@id\": \"https://schema.org/Event\"}",
+    "title": "Organization",
+    "description": "An event such as transformation, aggregation, commission etc.",
```
```suggestion
    "description": "An event such as transformation, aggregation, commission, etc.",
```

TallTed reviewed (changes requested):
Made many description strings more consistent with each other (start with article, close with period, etc.), particularly in ontology.  Some sample data is not good English (sentence fragments, primarily) but I'm not always sure what was intended, so some are now unchanged — but will want followup attention.

mkhraisha commented:
The remaining suggestions modify generated text, we probably need to actually dive into the generators and figure out how to fix them. This should probably be made its own issue and tracked accordingly. @OR13 @mprorock @TallTed

OR13 commented:
These proposed changed look good to me, @mkhraisha can you approve them / reject them so we can see where any gaps are remaining?

OR13 commented:
Feels like the generators may be getting in the way of having "realistic fixtures"... we should continue to discuss this on the related open issues.

TallTed commented:
Generated files should generally not be in git repos.  Whatever can be done to shift things so that the generated files go to github.io or similar would be best.

mkhraisha commented:
Apologies for the delays in tackling this. 
 Should be all good to go now.

OR13 commented:
This looks good to me.
